### DISCUSSION TITLE
How To Move Metadata from one stream to another

### DISCUSSION THREAD

**Discussion by Bobe001** [savonet/liquidsoap#3555] [User Support]
Hi I am putting together a script which receives a stream from a HTTP source. We then grab the metadata from this stream and send it to an Audio Processor, which strips off the metadata. When the output of the Audio Processor is entered into the script I want to inject the saved metadata but find that trying pretty much every know combination of commands for example: stream_ip = on_metadata(print_metadata,remote_input1) where remote_input1 is the original unprocessed stream and stream_ip is the processed stream. This and every other combination of commands I can fins will inject the metadata into the new stream but also injects the original audio as well. Is there a way of moving the metadata across 2 streams like this without having thew audio come across too?

**Comment by vitoyucepi** [comment]
Hi @Bobe001,
Have you tried replacing an audio track in the source?
https://www.liquidsoap.info/doc-2.2.2/multitrack.html

```ruby
a = input.http()
b = audioprocessor(a)
c = source(source.tracks(a).{audio=source.tracks(b).audio})
```
